File: python/heap-priority-queue/k-closest-points-to-origin.py
import heapq

class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        heap = []
        for i in points:
            heapq.heappush(heap, ((i[0] ** 2 + i[1] ** 2) ** 0.5, i))

        finalanswer = []
        for i in range(k):
            finalanswer.append(heapq.heappop(heap)[1])

        return finalanswer

File: c++/heap-priority-queue/k-closest-points-to-origin.cc
class Solution {
public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        std::priority_queue<std::pair<int, int>> pq;

        for (int i = 0; i < points.size(); ++i) {
            pq.push(std::make_pair(-(std::pow(points[i][0], 2) + std::pow(points[i][1], 2)), i));
        }

        std::vector<std::vector<int>> answer;
        for (int i = 0; i < k; ++i) {
            answer.push_back(points[pq.top().second]);
            pq.pop();
        }

        return answer;
    }
};
